### PR TITLE
[ADP-3288] Restructure integration suite to help HLS with multi home issues

### PR DESCRIPTION
**Pull request by paolino**
HLS is struggling with the current module layout for the integration test executable and test suite

- [x] Move other-modules needed for integration tests in integration library
- [x] Move also the Main there as it's a decent place to hole the common test definition (bad design with a small effect)

ADP-3288

### CODE REVIEW

File: 1 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Error: cabal: Unknown target '2'.
+There is no component '2'.
+The project has no package '2'.
+
+
```
Comment (HeinrichApfelmus) [inline 1]:
```suggestion
```

🤫

File: lib/wallet/integration/framework/Test/Integration/Run.hs
Comment (HeinrichApfelmus) [inline lib/wallet/integration/framework/Test/Integration/Run.hs:1]: Delete this file.

File: lib/wallet/test/integration/integration-tests.hs
Comment (HeinrichApfelmus) [inline lib/wallet/test/integration/integration-tests.hs:1]: Delete this file?

**Review by HeinrichApfelmus — APPROVED** [review]
Thanks! 😊 

Some files should not be there and need to be deleted. Looking good otherwise.

File: 1 (hunk 1 above)
Comment (paolino) [reply 1]: ah 2>1 ...

File: lib/wallet/test/integration/integration-tests.hs
Comment (paolino) [reply lib/wallet/test/integration/integration-tests.hs:1]: pointing to the same file from 2 stanzas (exe + test) is not liked from hls. That's why I factored it out in the lib. I will recheck in the future is they fix it, or it's cabal ? Let's see

**Pull request merged**